File: 385/b.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
  string s;
  cin >> s;

  vector<int> bears;
  int last = -4;
  while ((last = s.find("bear", last + 4)) != -1) {
    bears.push_back(last);
  }

  long long cnt = 0;

  for (int i = 0; i < s.size(); i++) {
    int li = lower_bound(bears.begin(), bears.end(), i) - bears.begin();
    if (li == bears.size()) {
      break;
    }

    int least = bears[li];
    int diff = (s.size() - least - 3);
    //cout << "least: " << least << " , " << diff << ":" << endl;
    cnt += diff;
  }

  cout << cnt << endl;
}

File: 385/b.py
import sys
import string

s = sys.stdin.readline().strip()

bears = []
last = -4
while True:
  last = string.find(s, "bear", last + 4)
  if (last == -1):
    break
  bears.append(last)

cnt = 0
for i in range(0, len(s) - 3):
  least = min([ x for x in bears if x >= i ] or [ -1 ])
  if least == -1:
    continue
  cnt += len([ x for x in range(least + 3, len(s)) ])

#print bears
print cnt
